ibis/backends/base/__init__.py

```diff
@@ -259,6 +259,8 @@ def to_pyarrow(
             A pyarrow table holding the results of the executed expression.
         """
         pa = self._import_pyarrow()
+        self._register_udfs(expr)
+        self._register_in_memory_tables(expr)
```

Comment:
We now have `_run_pre_execute_hooks` (currently in the base SQL backend, but not the base backend, but that'd be an easy fix), where backends can add hooks like this. We might move `_register_udfs` to be called from there in the base implementation. https://github.com/ibis-project/ibis/blob/ea1529d14aec891974076c4b093ac6c7c6bc9854/ibis/backends/base/sql/__init__.py#L263

Reply:
Refactored this and moved the methods to the base class.